util/options.cc

```diff
@@ -631,74 +631,29 @@ font_options_t::get_font (void) const
   hb_blob_t *blob = nullptr;
 
   /* Create the blob */
-  {
-    char *font_data;
-    unsigned int len = 0;
-    hb_destroy_func_t destroy;
-    void *user_data;
-    hb_memory_mode_t mm;
-
-    /* This is a hell of a lot of code for just reading a file! */
-    if (!font_file)
-      fail (true, "No font file set");
-
-    if (0 == strcmp (font_file, "-")) {
-      /* read it */
-      GString *gs = g_string_new (nullptr);
-      char buf[BUFSIZ];
+  if (!font_file)
+    fail (true, "No font file set");
+
+  if (0 == strcmp (font_file, "-")) {
+    /* read it */
+    GString *gs = g_string_new (nullptr);
+    char buf[BUFSIZ];
 #if defined(_WIN32) || defined(__CYGWIN__)
-      setmode (fileno (stdin), O_BINARY);
+    setmode (fileno (stdin), O_BINARY);
 #endif
-      while (!feof (stdin)) {
-	size_t ret = fread (buf, 1, sizeof (buf), stdin);
-	if (ferror (stdin))
-	  fail (false, "Failed reading font from standard input: %s",
-		strerror (errno));
-	g_string_append_len (gs, buf, ret);
-      }
-      len = gs->len;
-      font_data = g_string_free (gs, false);
-      user_data = font_data;
-      destroy = (hb_destroy_func_t) g_free;
-      mm = HB_MEMORY_MODE_WRITABLE;
-    } else {
-      GError *error = nullptr;
-      GMappedFile *mf = g_mapped_file_new (font_file, false, &error);
-      if (mf) {
-	font_data = g_mapped_file_get_contents (mf);
-	len = g_mapped_file_get_length (mf);
-	if (len) {
-	  destroy = (hb_destroy_func_t) g_mapped_file_unref;
-	  user_data = (void *) mf;
-	  mm = HB_MEMORY_MODE_READONLY_MAY_MAKE_WRITABLE;
-	} else
-	  g_mapped_file_unref (mf);
-      } else {
-	fail (false, "%s", error->message);
-	//g_error_free (error);
-      }
-      if (!len) {
-	/* GMappedFile is buggy, it doesn't fail if file isn't regular.
-	 * Try reading.
-	 * https://bugzilla.gnome.org/show_bug.cgi?id=659212 */
-        GError *error = nullptr;
-	gsize l;
-	if (g_file_get_contents (font_file, &font_data, &l, &error)) {
-	  len = l;
-	  destroy = (hb_destroy_func_t) g_free;
-	  user_data = (void *) font_data;
-	  mm = HB_MEMORY_MODE_WRITABLE;
-	} else {
-	  fail (false, "%s", error->message);
-	  //g_error_free (error);
-	}
-      }
+    while (!feof (stdin)) {
+      size_t ret = fread (buf, 1, sizeof (buf), stdin);
+      if (ferror (stdin))
+	fail (false, "Failed reading font from standard input: %s",
+	      strerror (errno));
+      g_string_append_len (gs, buf, ret);
     }
-
-    if (debug)
-      mm = HB_MEMORY_MODE_DUPLICATE;
```

Comment:
The only difference is removal of this. If it is vital, I can revert my change on options.cc and options.hh.

Reply:
Hmm, but I should see what should be done instead

Reply:
No this is fine.